### ISSUE TITLE
PerConnectionApolloHandler sends wrong sub-protocol

### ISSUE THREAD

**Issue by westito** [leangen/graphql-spqr-spring-boot-starter#147]
I tried subscribe with Postman but it fails with `Server sent no subprotocol`:

<img width="680" alt="Képernyőfotó 2024-04-27 - 16 39 11" src="https://github.com/leangen/graphql-spqr-spring-boot-starter/assets/536799/ed478cc2-d0d9-4f07-8a4d-564805e7d845">

I found that the client sends `graphql-transport-ws` sub-protocol but server only tests to `graphql-ws` because `getSubProtocols()` return only that. The official protocol is the `graphql-transport-ws`  as you see here: https://github.com/enisdenjo/graphql-ws/blob/master/PROTOCOL.md

**Comment by westito** [comment]
Meanwhile I realised the implemented protocol simply older than the currently recommended that Postman uses.

**Comment by kaqqao** [comment]
Yes, true. I figured there's no point in implementing the new protocol in SPQR since Spring itself already does. I'll release SPQR Starter 2.0.0.rc1 soon, possibly even today. The new version removes SPQR's entire communication layer and uses Spring directly. This will obviously be a breaking change, but it shouldn't cause too much trouble.
In the meantime, it's possible to already use Spring for communication with the current SPQR starter. I've been using it in production like that myself ever since Spring for GraphQL was released. It's pretty straightforward, I can post an example later.

**Comment by swwun** [comment]
> Yes, true. I figured there's no point in implementing the new protocol in SPQR since Spring itself already does. I'll release SPQR Starter 2.0.0.rc1 soon, possibly even today. The new version removes SPQR's entire communication layer and uses Spring directly. This will obviously be a breaking change, but it shouldn't cause too much trouble. In the meantime, it's possible to already use Spring for communication with the current SPQR starter. I've been using it in production like that myself ever since Spring for GraphQL was released. It's pretty straightforward, I can post an example later.

Hi @kaqqao, is there any update on this, by any chance? It would be great if there is an example to use Spring for communication with the current SPQR starter.
